go/vt/vtgate/engine/route.go

```diff
@@ -154,6 +159,9 @@ func (code RouteOpcode) MarshalJSON() ([]byte, error) {
 
 // Execute performs a non-streaming exec.
 func (route *Route) Execute(vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool) (*sqltypes.Result, error) {
+	if route.QueryTimeout != 0 {
+		vcursor.SetContextTimeout(route.QueryTimeout)
```

Comment:
this needs to get and `defer cancel()`. Same below.

Reply:
Oooh yes good point.

Reply:
oh good catch!